thirdeye/thirdeye-frontend/app/pods/components/rootcause-trend/component.js

```diff
@@ -0,0 +1,389 @@
+import {
+  toCurrentUrn,
+  toBaselineUrn,
+  toOffsetUrn,
+  hasPrefix,
+  filterPrefix,
+  toMetricLabel,
+  isInverse,
+  toColorDirection,
+  makeSortable
+} from 'thirdeye-frontend/utils/rca-utils';
+import {
+  humanizeChange,
+  humanizeFloat,
+  humanizeScore
+} from 'thirdeye-frontend/utils/utils';
+import moment from 'moment';
+import _ from 'lodash';
+
+const ROOTCAUSE_TREND_MAX_COLUMNS = 12;
+
+export default Component.extend({
+  classNames: ['rootcause-metrics'],
+
+  //
+  // external properties
+  //
+
+  /**
+   * Entities cache
+   * @type {object}
+   */
+  entities: null,
+
+  /**
+   * Metric aggregates
+   * @type {object}
+   */
+  timeseries: null,
+
+  /**
+   * Investigation context
+   * @type {object}
+   */
+  context: null,
+
+  /**
+   * User-selected urns
+   * @type {Set}
+   */
+  selectedUrns: null,
+
+  /**
+   * Callback on metric selection
+   * @type {function}
+   */
+  onSelection: null, // function (Set, state)
+
+  /**
+   * loading status for component
+   * @type {boolean}
+   */
+  isLoading: false,
+
+  //
+  // internal properties
+  //
+
+  /**
+   * Display flag for table. Hack for ember models table only re-rendering columns on init()
+   * @type {bool}
+   */
+  showTable: true,
+
+  /**
+   * Hack for ember models table only re-rendering columns on init()
+   */
+  _columnObserver: observer('columns', function () {
+    this.set('showTable', false);
+    later(this, () => this.set('showTable', true), 0);
+  }),
+
+  /**
+   * Start time for time table, overridden by drop down
+   * @type {int}
+   */
+  startTime: computed('context', function () {
+    return this.get('context').analysisRange[0];
+  }),
+
+  /**
+   * Start time formatted
+   * @type {string}
+   */
+  startTimeFormatted: computed('startTime', function () {
+    return this._formatTime(this.get('startTime'));
+  }),
+
+  /**
+   * Start time options for dropdown
+   */
+  startTimeOptions: computed('availableBuckets', function () {
+    const { availableBuckets } = this.getProperties('availableBuckets');
+
+    const options = [];
+    for (let i = 0; i < availableBuckets.length; i += ROOTCAUSE_TREND_MAX_COLUMNS) {
+      options.push(this._formatTime(moment(availableBuckets[i])));
+    }
+
+    return options;
+  }),
+
+  /**
+   * Reverse lookup mapping for start time options for dropdown
+   */
+  startTimeOptionsMapping: computed('availableBuckets', function () {
+    const { availableBuckets } = this.getProperties('availableBuckets');
+
+    const options = {};
+    for (let i = 0; i < availableBuckets.length; i += ROOTCAUSE_TREND_MAX_COLUMNS) {
+      options[this._formatTime(moment(availableBuckets[i]))] = availableBuckets[i];
+    }
+
+    return options;
+  }),
+
+  /**
+   * Possible time buckets from analysis range
+   * @type {int[]}
+   */
+  availableBuckets: computed('context', function () {
+    const { context } = this.getProperties('context');
+
+    const buckets = [];
```

Comment:
This should be a let if we decide to mutate the variable's value.

let buckets = [];

Reply:
@aaronucsd I'm fine with `const` here since the reference doesn't change. Is it best practice to use a `let` to explicitly indicate that the array content might change?

Reply:
food for thought: https://mathiasbynens.be/notes/es6-const

Reply:
@justYves Yes. There are two factors for this. One is it is correct the reference default type (array) assignment isn't going to change (in this current case). Actually it's content/children is changing is the reason why this const is still valid with the linter. 

```
const foo = {};
foo.bar = 42;
console.log(foo.bar);
```

However, we can be blinded to see future reference changes if this were to be passed to a method later that does mutate the original reference. 

```
const foo = {};
someMethodChangesFoo(foo);//Uncaught TypeError: Assignment to constant variable.
console.log(foo);
```

Reply:
The latter one does not mutate to original "reference" foo, but the object that is referenced by by "foo"

Reply:
HI @apucher.  True, as the reference of the param is to the object and not the original handler. My point is later in your code, when you decide to extract this out to a method (util/service etc), someone can accidentally modify it as such `foo = {a:42};` where it will just needs to update to `let`.  For now it's fine, just leave it as const. Thanks.